content/fast/incorporate-performance-budgets-into-your-build-tools/codelab-setting-performance-budgets-with-webpack.md

```diff
@@ -0,0 +1,193 @@
+    seen before. 1 “KiB” and 1 “KB” are fairly close in size: 1 KiB is 1024 bytes, 
+    while 1 KB is 1000 bytes.)
+</div>
+
+![Webpack output showing bundle size of 323 KiB](webpack-bundle-sizes.png)
+
+These warnings are enabled by default in [production mode](https://webpack.js.org/concepts/mode/)
+and the default threshold is **244 KiB uncompressed**, for both assets and
+[entry points](https://webpack.js.org/concepts/entry-points/)
+(the combination of all assets used during the initial load of a page).
+
+<figure>
+  <img src="./webpack-warning.png" alt="Webpack warning that the asset exceeds the recommended size limit">
+  <figcaption>
+    Webpack warning you about bulky JS bundle ⚠️
+  </figcaption>
+</figure>
+
+Webpack will not only warn you, but it will also give you a recommendation on
+how to downsize your bundles. You can learn more about the recommended techniques on
+[Web Fundamentals](https://developers.google.com/web/fundamentals/performance/webpack/use-long-term-caching#lazy-loading).
+
+<figure> 
+    <img src="webpack-performance-recommendation.png" alt="Webpack performance optimization recommendation" class="screenshot"> 
+    <figcaption>Webpack performance optimization recommendation 💁</figcaption> 
+</figure>
+
+## Set a custom performance budget
+
+An appropriate performance budget will depend on the nature of your project.
+It’s always best to [do your own research](https://web.dev/fast/your-first-performance-budget).
+A [good rule](https://web.dev/fast/your-first-performance-budget#budget-for-quantity-based-metrics)
+of thumb is to deliver under 170 KB of compressed/minified
+[critical-path](https://developers.google.com/web/fundamentals/performance/critical-rendering-path/)
+resources.
+
+<div class="aside note">
+    Critical-path resources are the minimal set of resources required
+    for the browser to display the first screen's worth of content.
+</div>
+
+For this simple demo, try being even more conservative and set the budget to
+100 KB. In the configuration file, add the following:
+
+<pre class="prettyprint"> 
+    module.exports = { 
+        //... 
+            performance: { 
+            maxAssetSize: 100000, 
+            maxEntrypointSize: 100000, 
+            hints: 'warning' 
+        } 
+    };
+</pre>
+
+The new performance budget is set in **bytes**:
+
+- 100000 bytes for individual assets (maxAssetSize)
+
+- 100000 bytes for the entry-point (maxEntrypointSize)
+
+In this case, there’s only one bundle which also acts as the entry point.
+
+Possible values for **hints** are:
+
+1. **`warning`** (default): Displays a yellow warning message, but the build
+   passes. It’s best to use this in development environments.
+
+2. **`error`**: Displays a red error message, but the build still passes. This
+   setting is recommended for production builds.
+
+3. **`false`**: No warnings or errors are shown.
+
+<figure> 
+    <img src="webpack-error.png" alt="" class="screenshot"> 
```

Comment:
I recently came across [this article](https://www.scottohara.me/blog/2019/01/21/how-do-you-figure.html) which covers the proper use of figure/figcaption.

The tl;dr is that we should put alt text on our images. There's more we can do in the future, but for now I think making sure we have alt text that describes the content of the image is useful enough.